### PR TITLE
Completed T13

### PR DESCRIPTION
**Pull request by Ajay07Metamation**

### CODE REVIEW

File: Training/Program.cs [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿// Program for checking Valid Password 
+using static System.Console;
+Write ("Enter your Password : ");
+string pass = Console.ReadLine () ?? "";
+WriteLine (valid (pass));
+String valid (string s) {
```
Comment (rajeevramanTMM) [inline Training/Program.cs:6]: Use string data type not the class for variable and function return type..

File: Training/Program.cs [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+   String result = "";
+   String spl = "!@#$%^&*()-+";
+   if (s.Length >= 6) {
+      if (!s.Any (char.IsDigit)) result += "Password should contain atleast 1 numerical \n";
+      if (!s.Any (char.IsUpper)) result += "Password should contain atleast 1 Upper case \n";
+      if (!s.Any (char.IsLower)) result += "Password should contain atleast 1 Lower case \n";
+      if (!s.Any (spl.Contains)) result += "Password should contain atleast 1 special character \n";
```
Comment (rajeevramanTMM) [inline Training/Program.cs:13]: You have to show the special character to the user if not he cant create a password.

File: Training/Program.cs [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
+﻿// Program for checking Valid Password 
+using static System.Console;
+Write ("Enter your Password : ");
+string pass = Console.ReadLine () ?? "";
+WriteLine (valid (pass));
+String valid (string s) {
+   String result = "";
+   String spl = "!@#$%^&*()-+";
+   if (s.Length >= 6) {
+      if (!s.Any (char.IsDigit)) result += "Password should contain atleast 1 numerical \n";
```
Comment (rajeevramanTMM) [inline Training/Program.cs:10]: Multiple times the same message is repeated can you clean up the code.

File: Training/Program.cs [hunk 4]
```diff
@@ -1,7 +1,17 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿// Program for checking Valid Password 
+using static System.Console;
+Write ("Enter your Password : ");
+string pass = Console.ReadLine () ?? "";
```
Comment (rajeevramanTMM) [inline Training/Program.cs:4]: Remove console key word

File: Training/Program.cs (hunk 1 above)
Comment (rajeevramanTMM) [inline Training/Program.cs:6]: Provide a valid function name with capital letter for eg: IsValid.

**Review by rajeevramanTMM — CHANGES_REQUESTED** [review]

File: Training/Program.cs (hunk 1 above)
Comment (Ajay07Metamation) [reply Training/Program.cs]: All changes done in latest commit

File: Training/Program.cs [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
+WriteLine (IsValid (pass));
+string IsValid (string s) {
+   string result = "";
+   string spl = "!@#$%^&*()-+";
+   if (s.Length >= 6) {
+      var res = "Password should contain atleast 1 ";
+      if (!s.Any (char.IsDigit)) result += res + "Numerical \n";
+      if (!s.Any (char.IsUpper)) result += res + "Upper case \n";
+      if (!s.Any (char.IsLower)) result += res + "Lower case \n";
+      if (!s.Any (spl.Contains)) result += res + "Special character !@#$%^&*()-+ ";
```
[the remark this answers is not in the document]
Comment (Ajay07Metamation) [reply Training/Program.cs]: All changes done in latest commit

**Pull request closed without merging**